tpm/tpm.go

```diff
@@ -713,6 +714,104 @@ func OwnerReadSRK(rw io.ReadWriter, ownerAuth digest) ([]byte, error) {
 	return tpmutil.Pack(pk)
 }
 
+// ReadEKCert reads the EKCert from the NVRAM
+func ReadEKCert(rw io.ReadWriter, ownAuth digest) ([]byte, error) {
+	var (
+		data     []byte
+		ekbuf    []byte
+		buff     io.Reader
+		offset   uint32
+		bufSize  uint32
+		tag      uint16
+		certType uint8
+		certSize uint16
+	)
+	const certIndex = 0x1000f000
+	offset = 0
+	data, err := NVReadValue(rw, certIndex, offset, 5, ownAuth)
+	if err != nil {
+		return nil, err
+	}
+	offset = offset + 5
+	buff = bytes.NewReader(data)
+
+	if err := binary.Read(buff, binary.BigEndian, &tag); err != nil {
```

Comment:
this can be

```
var header struct {
    tag uint16
    certType uint8
    certSize uint16
}
if err := binary.Read(buff, binary.BigEndian, &header); err != nil {
    // handle error
}
```